### ISSUE TITLE
No error for extra semicolon in the end of function body

### ISSUE THREAD

**Issue by alekseykulikov** [eslint/eslint#7853]
**Tell us about your environment**

* **ESLint Version:** v3.12.2
* **Node Version:** v6.9.2
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
{
  "rules": {
    "semi": [2, "never"]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
function foo () {
  return 'bar'
};
```

**What did you expect to happen?**
I expect to see error: `Extra semicolon` triggered by `semi` rule for line `3`.

**What actually happened? Please include the actual, raw output from ESLint.**
No errors.

**Comment by platinumazure** [comment]
The `semi` rule only triggers for semicolons that are between statements (such that a semicolon or ASI is required by the language grammar). After a function declaration, a semicolon is never required. 

We have a separate rule for catching completely optional semicolons, `no-extra-semi`. Turn that on and it should catch the semicolon on line 3.

**Comment by alekseykulikov** [comment]
Tank you @platinumazure it helped!
